docs/src/pages/playground.js

```diff
@@ -185,15 +185,29 @@ export function DatalogDemo() {
 }
 
 export function EditorDemo({ compileFn, examples, mermaidId }) {
+  if (siteConfig.customFields.LOAD_PLAYGROUND !== '1') {
+    return <div>Please set LOAD_PLAYGROUND environment variable to 1 to enable the playground.</div>;
+  }
+
   const [program, setProgram] = useState(Object.values(examples)[0]);
   const [showingMermaid, setShowingMermaid] = useState(true);
   const [editorAndMonaco, setEditorAndMonaco] = useState(null);
 
-  if (siteConfig.customFields.LOAD_PLAYGROUND !== '1') {
-    return <div>Please set LOAD_PLAYGROUND environment variable to 1 to enable the playground.</div>;
-  }
+  const [showGraphOpts, setShowGraphOpts] = useState(false);
+  const [writeGraphConfig, setWriteGraphConfig] = useState({
+    noSubgraphs: false,
+    noVarnames: false,
+    noPullPush: false,
+    noHandoffs: false,
+    opShortText: false,
+  });
+  const writeGraphConfigOnChange = (name) => {
+    writeGraphConfig[name] = !writeGraphConfig[name];
+    setWriteGraphConfig({ ...writeGraphConfig });
+    return true;
+  };
 
-  const { output, diagnostics } = (compileFn)(program);
+  const { output, diagnostics } = (compileFn)(program, ...Object.values(writeGraphConfig));
```

Comment:
Does JS guarantee that the order of keys is preserved? Not sure myself...

Reply:
good question

Reply:
Yes but maybe worth making it less janky

Reply:
Left it janky but added some comments